src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeParse.cs

```diff
@@ -3299,25 +3300,56 @@ private static bool MatchAbbreviatedMonthName(ref __DTString str, DateTimeFormat
             result = -1;
             if (str.GetNext())
             {
-                //
-                // Scan the month names (note that some calendars has 13 months) and find
-                // the matching month name which has the max string length.
-                // We need to do this because some cultures (e.g. "cs-CZ") which have
-                // abbreviated month names with the same prefix.
-                //
-                int monthsInYear = (dtfi.GetMonthName(13).Length == 0 ? 12 : 13);
-                for (int i = 1; i <= monthsInYear; i++)
+                if (ReferenceEquals(dtfi, DateTimeFormat.InvariantFormatInfo))
```

Comment:
nit:
would make sense to make the [DateTimeFormatInfo.CultureName](https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormatInfo.cs,153) be internal and just check this name if it is empty string? 

I know you are covering the main stream case. Checking the name can also help if someone creating InvariantCulture from the name.

Reply:
Checking the culture name will cover the cases when using `CultureInfo.InvariantCulture` and the case when doing `new CultureInfo("")`. Your change covers the first case which is the mainstream case. The latter case is rare and that is why I mentioned my comment as `nit`. If you think checking the name will not add any cost than `ReferenceEquals` then it will not hurt doing it.